### ISSUE TITLE
Project API add avatar

### ISSUE THREAD

**Issue by jflefebvre06** [gmessner/gitlab4j-api#284]
Hello,

Could you please add a method to add avatar to projet. Below a sample can help you

```java
public Project addProjectAvatar(final Project project, final File fileToUpload) throws GitLabApiException {
        Validate.notNull(project, "project cannot be null");
        Validate.notNull(fileToUpload, "fileToUpload cannot be null");
        Validate.isTrue(fileToUpload.isFile(), "fileToUpload is not a file");
        try {
            final MultipartEntityBuilder builder = MultipartEntityBuilder.create();
            builder.setMode(HttpMultipartMode.BROWSER_COMPATIBLE);
            builder.addBinaryBody("avatar", fileToUpload, ContentType.DEFAULT_BINARY, fileToUpload.getName());
            // Si on ne passe que l'avatar on obtien une erreur
            builder.addTextBody("name", project.getName(), ContentType.DEFAULT_BINARY);
            final HttpPut put = new HttpPut(this.gitLabApi.getGitLabServerUrl() + this.gitLabApi.getApiVersion().getApiNamespace() + "/projects/" + project.getId());
            put.setHeader("PRIVATE-TOKEN", this.gitLabApi.getAuthToken());
            put.setEntity(builder.build());
            // FIXME Gestion du proxy
            final HttpClient client = HttpClientBuilder.create().build();
            final HttpResponse response = client.execute(put);
            final int statusCode = response.getStatusLine().getStatusCode();
            final HttpEntity responseEntity = response.getEntity();
            ...
        } catch (final IOException ioException) {
            throw new GitLabApiException(ioException);
        }
    }
```

**Comment by gmessner** [comment]
@jflefebvre06 
I've added ```setProjectAvatar()``` to the ```ProjectApi``` which is included in release 4.9.7

Please verify and close when you have a chance.

**Comment by jflefebvre06** [comment]
Works fine thank you

**Comment by jflefebvre06** [comment]
Warning setting avatar :

```console
déc. 20, 2018 8:45:38 AM org.glassfish.jersey.message.internal.HeaderUtils checkHeaderChanges
AVERTISSEMENT: There are some request headers that have not been sent by connector [org.glassfish.jersey.apache.connector.ApacheConnector]. Probably you added those headers in WriterInterceptor or MessageBodyWriter. That feature is not supported by the connector. Please, do not modify headers in WriterInterceptor or MessageBodyWriter or use default HttpUrlConnector instead.
Unsent header changes: [MIME-Version, Content-Type]
déc. 20, 2018 8:45:50 AM org.glassfish.jersey.message.internal.HeaderUtils checkHeaderChanges
AVERTISSEMENT: There are some request headers that have not been sent by connector [org.glassfish.jersey.apache.connector.ApacheConnector]. Probably you added those headers in WriterInterceptor or MessageBodyWriter. That feature is not supported by the connector. Please, do not modify headers in WriterInterceptor or MessageBodyWriter or use default HttpUrlConnector instead.
Unsent header changes: [MIME-Version, Content-Type]
déc. 20, 2018 8:46:11 AM org.glassfish.jersey.message.internal.HeaderUtils checkHeaderChanges
AVERTISSEMENT: There are some request headers that have not been sent by connector [org.glassfish.jersey.apache.connector.ApacheConnector]. Probably you added those headers in WriterInterceptor or MessageBodyWriter. That feature is not supported by the connector. Please, do not modify headers in WriterInterceptor or MessageBodyWriter or use default HttpUrlConnector instead.
Unsent header changes: [MIME-Version, Content-Type]
```

**Comment by gmessner** [comment]
@jflefebvre06 
I'm not able to replicate this warning, what version of Jersey are you using?

**Comment by jflefebvre06** [comment]
I'am not overriding jersey version

![image](https://user-images.githubusercontent.com/2685607/50347339-838d6c80-0535-11e9-92b0-197ce0e29f4d.png)

**Comment by gmessner** [comment]
@jflefebvre06 
I've simplified the code for doing the upload, my hope is it took care of the warning you are experiencing. Since I am not getting the warning I could not verify, could you test using 4.9.10 and let me know?

**Comment by jflefebvre06** [comment]
Hello,

Same problem with 4.10 and 4.11, to reproduce, you must add a proxy (https://stackoverrun.com/fr/q/8791791).

```
Warning 
Do not use ApacheConnectorProvider nor GrizzlyConnectorProvider neither JettyConnectorProvider connector implementations with Jersey Multipart features. See Header modification issue warning for more details. 
```

After adding proxy you will get a 400 status code if you use setProjectAvatar or uploadFile.


```java
try (MultiPart multiPart = new FormDataMultiPart()) {
    multiPart.bodyPart(filePart);

    final ClientConfig clientConfig = new ClientConfig();
    clientConfig.connectorProvider(new ApacheConnectorProvider());
    clientConfig.property(ClientProperties.PROXY_URI, "http://xxx:8080");
    clientConfig.register(JacksonJson.class);
    clientConfig.register(MultiPartFeature.class);

    final ClientBuilder clientBuilder = ClientBuilder.newBuilder().withConfig(clientConfig);
    
    final WebTarget target = clientBuilder.build().target(this.gitLabApi.getGitLabServerUrl() + this.gitLabApi.getApiVersion().getApiNamespace() + "/projects/" + project.getId() + "/uploads")
            .property(ClientProperties.FOLLOW_REDIRECTS, true);

    final Response response = target.request().header("PRIVATE-TOKEN", this.gitLabApi.getAuthToken()).accept("application/json").post(Entity.entity(multiPart, multiPart.getMediaType()));

    System.out.println("status : " + response.getStatus()); // <--------- 400

    final FileUpload fileUpload = response.readEntity(FileUpload.class);

    System.out.println("fileUpload : " + fileUpload);

            }
```



The solution : wrap `multiPart.getMediaType()` with `Boundary.addBoundary(multiPart.getMediaType())`

```java
try (MultiPart multiPart = new FormDataMultiPart()) {
    multiPart.bodyPart(filePart);

    final ClientConfig clientConfig = new ClientConfig();
    clientConfig.connectorProvider(new ApacheConnectorProvider());
    clientConfig.property(ClientProperties.PROXY_URI, "http://xxx:8080");
    clientConfig.register(JacksonJson.class);
    clientConfig.register(MultiPartFeature.class);

    final ClientBuilder clientBuilder = ClientBuilder.newBuilder().withConfig(clientConfig);
    
    final WebTarget target = clientBuilder.build().target(this.gitLabApi.getGitLabServerUrl() + this.gitLabApi.getApiVersion().getApiNamespace() + "/projects/" + project.getId() + "/uploads")
            .property(ClientProperties.FOLLOW_REDIRECTS, true);

    final Response response = target.request().header("PRIVATE-TOKEN", this.gitLabApi.getAuthToken()).accept("application/json")
                        .post(Entity.entity(multiPart, Boundary.addBoundary(multiPart.getMediaType())));
    System.out.println("status : " + response.getStatus()); // <--------- 201

    final FileUpload fileUpload = response.readEntity(FileUpload.class);

    System.out.println("fileUpload : " + fileUpload);

            }
```

**Comment by jflefebvre06** [comment]
Yes with the belowed code I still getting the warning

**Comment by jflefebvre06** [comment]
Sorry I click the "close and comment" :)

**Comment by gmessner** [comment]
@jflefebvre06 
I have not found a way to take care of the warning, but I have a fix that will work when using a proxy.  Will this be sufficient?

**Comment by jflefebvre06** [comment]
Thank you greg yes it is sufficient

**Comment by gmessner** [comment]
@jflefebvre06 
The fix has been released in 4.9.12, please verify and close when you have time to do so.

**Comment by jflefebvre06** [comment]
Works fine thank you
